src/controllers/controlpickermenu.cpp

```diff
@@ -813,52 +791,43 @@ ControlPickerMenu::ControlPickerMenu(QWidget* pParent)
 ControlPickerMenu::~ControlPickerMenu() {
 }
 
-void ControlPickerMenu::addControl(QString group, QString control, QString title,
-                                   QString description,
-                                   QMenu* pMenu,
-                                   bool addReset) {
-    QAction* pAction = pMenu->addAction(title, &m_actionMapper,
-                                        SLOT(map()));
-    m_actionMapper.setMapping(pAction, m_controlsAvailable.size());
-    addAvailableControl(ConfigKey(group, control), title, description);
+void ControlPickerMenu::addSingleControl(QString group, QString control,
+                                         QString title, QString description,
+                                         QMenu* pMenu, QString prefix,
+                                         QString actionTitle) {
+    int controlIndex = m_controlsAvailable.size();
 
-    if (addReset) {
-        QString resetDescription = QString("%1 (%2)").arg(title,
-                                                          m_resetStr);
-        QString resetControl = QString("%1_set_default").arg(control);
-        QAction* pResetAction = pMenu->addAction(resetDescription,
-                                                 &m_actionMapper, SLOT(map()));
-        m_actionMapper.setMapping(pResetAction, m_controlsAvailable.size());
-        addAvailableControl(ConfigKey(group, resetControl),
-                            resetDescription, resetDescription);
+    if (actionTitle.isEmpty()) {
+        actionTitle = title;
     }
-}
 
-void ControlPickerMenu::addPrefixedControl(QString group, QString control,
-                                           QString controlTitle,
-                                           QString controlDescription, QString descriptionPrefix,
-                                           QMenu* pMenu, bool addReset) {
-    QAction* pAction = pMenu->addAction(controlTitle, &m_actionMapper, SLOT(map()));
-    m_actionMapper.setMapping(pAction, m_controlsAvailable.size());
+    auto pAction = std::make_unique<QAction>(actionTitle, pMenu);
```

Comment:
The lifetime of QAction is managed by the parent `pMenu`. Why do we need to use a unique pointer here?

Reply:
Ok, I removed the `std::unique_ptr`.